code/datums/helper_datums/map_template.dm

```diff
@@ -83,16 +83,31 @@
 		map_templates[T.name] = T
 
 	preloadRuinTemplates()
+	preloadShuttleTemplates()
 
 /proc/preloadRuinTemplates()
-	var/list/potentialSpaceRuins = generateMapList(filename = "_maps/RandomRuins/SpaceRuins/_maplisting.txt", blacklist = "config/spaceRuinBlacklist.txt")
-	for(var/ruin in potentialSpaceRuins)
-		var/datum/map_template/T = new(path = "[ruin]", rename = "[ruin]")
-		space_ruins_templates[T.name] = T
-		map_templates[T.name] = T
+	// Still supporting bans by filename
+	var/list/banned = generateMapList("config/lavaRuinBlacklist.txt")
+	banned += generateMapList("config/spaceRuinBlacklist.txt")
+
+	for(var/item in subtypesof(/datum/map_template/ruin))
+		var/datum/map_template/ruin/ruin_type = item
+		// screen out the abstract subtypes
+		if(!initial(ruin_type.id))
+			continue
+		var/datum/map_template/ruin/R = new ruin_type()
```

Comment:
It may not be worth it but you can avoid creating a new datum for every ruin by using initial() to get the var values for these